app/javascript/chat/ReportAbuse/index.jsx

```diff
@@ -0,0 +1,146 @@
+import { h } from 'preact';
+import PropTypes from 'prop-types';
+import { useState } from 'preact/hooks';
+import { reportAbuse, blockUser } from '../actions/requestActions';
+import { addSnackbarItem } from '../../Snackbar';
+import { Button } from '@crayons';
+
+function ReportAbuse({ resource: data, closeReportAbuseForm }) {
+  const [category, setCategory] = useState(null);
+
+  const handleChange = (e) => {
+    setCategory(e.target.value);
+  };
+
+  const handleSubmit = async () => {
+    const response = await reportAbuse(
+      data.message,
+      'connect',
+      category,
+      data.user_id,
+    );
+    const { success, message } = response;
+    if (success) {
+      const confirmBlock = window.confirm(
+        `Are you sure you want to block this person? This will:
+      - prevent them from commenting on your posts
+      - block all notifications from them
+      - prevent them from messaging you via DEV Connect`,
+      );
+      if (confirmBlock) {
+        const response = await blockUser(data.user_id);
+        if (response.result === 'blocked') {
+          addSnackbarItem({
+            message: 'Your report has been submitted and User has been blocker',
+          });
+        }
+      } else {
+        addSnackbarItem({ message: 'Your report has been submitted.' });
+      }
+      closeReportAbuseForm();
+    } else {
+      addSnackbarItem({ message });
+    }
+  };
+
+  return (
+    <div className="p4">
+      <div className="p-4 grid gap-2 crayons-card mb-4">
```

Comment:
refactor (blocking): These sections of content should use the [section](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/section) element instead.